### PR TITLE
feat: tools-2777 associate validation errors with the config section that caused…

### PR DESCRIPTION
**Pull request by dwelch-spike**
… them
Note: tools-2885 was found while working on this PR so I fixed it in https://github.com/aerospike/asconfig/pull/38/commits/c0dfad673275453def42c54a4411cf2e720f7fd7 . The issue caused the output of the validate command to go to stderr.

I ended up backing the tools-2885 changes out.

### CODE REVIEW

File: conf/config_validator.go
Comment (jdogmcsteezy) [inline conf/config_validator.go:159]: The only possible edge case I can think of is the `logging` context. In the schema `logging` is an array. Will this work as expected in that case?

**Review by jdogmcsteezy — CHANGES_REQUESTED** [review]

File: conf/config_validator.go
Comment (dwelch-spike) [reply conf/config_validator.go:159]: `logging` works, see below
```shell
./bin/asconfig validate -a 6.4.0 testdata/cases/server70/server70.conf
context: logging.console
        - description: Additional property drv-mem is not allowed, error-type: additional_property_not_allowed
```
logging.0 goes to logging.console in this case. It works because the function only expects the json config to be a slice at the point where an index is found. When `key` == "logging" jsonConfig is the map containing "logging", then jsonConfig is set the logging slice, and key becomes "0" at which point jsonConfig is asserted as a slice.

File: cmd/root.go
Comment (dwelch-spike) [reply cmd/root.go]: got it, changed

**Comment by dwelch-spike** [comment]
I found that the validation failures the validate command prints were actually going to stderr not stdout. They should be going to stdout. I filed a bug ticket https://aerospike.atlassian.net/browse/TOOLS-2885, since it was a 1 line change I fixed the issue in https://github.com/aerospike/asconfig/pull/38/commits/c0dfad673275453def42c54a4411cf2e720f7fd7 . Just changing cmd.Print to fmt.Print does the trick.

**Review by jdogmcsteezy — APPROVED** [review]
LGTM after TOOLS-2885 is backed out

**Comment by dwelch-spike** [comment]
tools-2885 is backed out, waiting for tests to pass then will merge

**Pull request merged**
